docs2/site/docs/migrations/migration8.md

```diff
@@ -389,6 +389,24 @@ property to `true` for all object graph types to retain the existing behavior.
 
 Allows to revisit the schema after all other methods (types/fields/etc) have been visited.
 
+### 13. OneOf Input Object support added
+
+OneOf Input Objects are a special variant of Input Objects where the type system
+asserts that exactly one of the fields must be set and non-null, all others
+being omitted. This is useful for representing situations where an input may be
+one of many different options.
+
+See: https://github.com/graphql/graphql-spec/pull/825
+
+To use this feature:
+- **Code-First**: Set the `IsOneOf` property on your `InputObjectGraphType` to `true`.
+- **Schema-First**: Use the `@oneOf` directive on the input type in your schema definition.
+- **Type-First**: Use the `[OneOf]` directive on the CLR class.
+
+Note: the feature is still a draft and has not made it into the official GraphQL spec yet.
+It is expected to be added once it has been implemented in multiple libraries and proven to be useful.
+It is not expected to change from the current draft.
```

Comment:
Should it be behind the experimental switch then?

Reply:
I wouldn't.  They're going to merge it into the released GraphQL version now (see https://github.com/graphql/graphql-spec/pull/825#issuecomment-2148072272 ), and besides, so long as a user doesn't use the feature, then it's like it doesn't exist.  The only exception is if someone defined `@oneOf` in their own schema-first schema.  In which case it's going to break their schema soon enough.  But even if a code-first user defined their own `@oneOf`, it won't change anything because the flag on the input type won't be set.

Reply:
The feature, in its current state, is already being merged into numerous implementations.

- https://github.com/graphql/graphql-spec/pull/825#issuecomment-2135724148
- https://github.com/graphql/graphql-spec/pull/825#issuecomment-2143721271

Reply:
Oh, and introspecting `__Type` will show a new field.